v3/src/components/case-tile-common/attribute-menu/attribute-menu-list.tsx

```diff
@@ -113,7 +117,23 @@ const AttributeMenuListComponent = forwardRef<HTMLDivElement, IProps>(
       }
     },
     {
-      itemKey: "DG.TableController.headerMenuItems.recoverFormula"
+      itemKey: "DG.TableController.headerMenuItems.recoverFormula",
+      isEnabled: () => !metadata?.isEditProtected(attributeId) &&
+          !attribute?.hasFormula &&
+          !!metadata?.getAttributeDeletedFormula(attributeId),
+      handleClick: () => {
+        data?.applyModelChange(() => {
+          if (!metadata) {
+            console.warn(`Metadata not found for data set: ${data?.id}`)
+            return
+          }
+          metadata.recoverAttributeFormula(attributeId)
+        }, {
+          undoStringKey: "DG.Undo.caseTable.editAttributeFormula",
+          redoStringKey: "DG.Undo.caseTable.editAttributeFormula",
+          log: logMessageWithReplacement("Recover formula for attribute %@", { name: attribute?.name })
```

Comment:
Out of curiosity, did you check to see whether this is the message v2 issued? We've been trying to maintain compatibility in case there are clients out there that care.

Reply:
No the log messages do not seem to match v2. I looked around and it does seem like most of the other log messages do match. So I'll try to update these.

This is what is used in v2 for recovering the formula
```
          var action = "attributeEditFormula";
          this.log = "%@: { name: '%@', collection: '%@', formula: '%@' }".fmt(
              action, tAttrName, tCollection.get('name'), tFormula);
```
where `tFormula` is the deleted formula.

And this is what is used when deleting the formula
```
          var action = "attributeEditFormula";
          this.log = "%@: { name: '%@', collection: '%@', formula: '%@' }".fmt(
              action, tAttrName, tCollection.get('name'), tFormula);
```
where `tFormula` is `''`

Reply:
I will update this PR so the log messages match for the main actions deleting a formula and recovering a formula. I'm not going to bother with the log messages that happen during Undo and Redo. In v2 these messages are customized for these actions as well.